### PR TITLE
feat: Port webhooks api to new switch pattern

### PR DESCRIPTION
**Pull request by rishtigupta**
# Port webhooks api to new switch pattern 

## Description:
I had to comment out the example code for webhooks since the CI kept failing for it. 
Will publish this sdk version and update the tests accordingly in a follow-up pr.

## Issue
Closes https://github.com/momentohq/dev-eco-issue-tracker/issues/956

### CODE REVIEW

**Comment by rishtigupta** [comment]
Will update the integ tests and examples in a separate pr after this pr is merged.

File: packages/common-integration-tests/src/webhooks.ts [hunk 1]
```diff
@@ -84,10 +84,6 @@ export function runWebhookTests(
           expect(webhookWeAreLookingFor).toBeTruthy();
         } else if (resp instanceof ListWebhooks.Error) {
           throw new Error(`list webhooks request failed: ${resp.message()}`);
-        } else {
-          throw new Error(
-            `unknown error occured when making a 'listWebhooks' request: ${resp.toString()}`
-          );
```
Comment (cprice404) [inline packages/common-integration-tests/src/webhooks.ts:90]: what's the motivation to remove these?

**Comment by cprice404** [comment]
can you update the pr title / commit message before merging please? thanks!

File: packages/common-integration-tests/src/webhooks.ts (hunk 1 above)
Comment (rishtigupta) [reply packages/common-integration-tests/src/webhooks.ts:90]: They were giving me warning saying these paths are redundant and will never be reached.

Comment (cprice404) [reply packages/common-integration-tests/src/webhooks.ts:90]: wow, that's cool.

**Review by cprice404 — APPROVED** [review]

**Pull request merged**